frontend/src/components/WeatherData.jsx

```diff
@@ -0,0 +1,19 @@
+export default function WeatherData({ weather }) {
+  function formatTemp(t) {
+    const sign = t > 0 ? "+" : ""
+    return sign + Math.round(t * 10) / 10 + " °C"
+  }
+
+  function formatWind(ws) {
+    return Math.round(ws * 10) / 10 + " m/s"
+  }
+
+  return (
+    <div className="data">
+      <div>Temp: {formatTemp(weather.temp)}</div>
+      <div>{weather.rain ? "Rain" : "No rain"}</div>
+      <div>Humidity: {Math.floor(weather.humidity)} %</div>
```

Comment:
Looks a bit odd to me that we're flooring the humidity when we're rounding the other values. Also we have formatting functions for two of the other ones and do inline formatting for this one.